### ISSUE TITLE
old version of s3fs being installed (0.4.2)

### ISSUE THREAD

**Issue by scottyhq** [pangeo-data/pangeo-docker-images#165]
Cross posting https://discourse.pangeo.io/t/slow-xarray-open-zarr-with-default-s3fs-0-4-2-fixed-in-latest-0-5-1/1073 

Our [unpinned environment specification](https://github.com/pangeo-data/pangeo-docker-images/blob/master/pangeo-notebook/environment.yml) included both `boto3` and `s3fs` which apparently resolves to installing s3fs 0.4.2

cc @emiliom and @martindurant or @ocefpaf for package dependency insights?

i can try 1) pinning s3fs>0.5, 2) removing boto3 from the requirement file (but i think it comes along with the awscli) and installing AWSCLI v2 as recommended here https://docs.aws.amazon.com/cli/latest/userguide/welcome-versions.html

**Comment by rabernat** [comment]
Thanks for catching this Scott! Given the pace of development on zarr / xarray / fsspec, it's really important we always have the latest release in our image.

**Comment by emiliom** [comment]
> 1) pinning s3fs>0.5

This seems like the best option for now. I don't know enough about where boto3 is used, so it seems safer to keep it.

**Comment by TomAugspurger** [comment]
For background, s3fs *used* to depend on boto3. More recent versions dropped that dependency (relying on botocore and aiobotocore instead).

I don't know if this is a conda solver failure or not. It's interesting that both conda and mamba apparently get the same result.

**Comment by scottyhq** [comment]
I just did a little digging around and see the issue, s3fs>=0.5 added the `aiobotocore` dependency:
```
s3fs 0.5.0 py_0
---------------
file name   : s3fs-0.5.0-py_0.tar.bz2
name        : s3fs
version     : 0.5.0
build       : py_0
build number: 0
size        : 22 KB
license     : BSD 3-Clause
subdir      : noarch
url         : https://conda.anaconda.org/conda-forge/noarch/s3fs-0.5.0-py_0.tar.bz2
md5         : 6075d8fa09227dbb6a2287ab11c6a245
timestamp   : 2020-08-22 19:41:23 UTC
dependencies: 
  - aiobotocore
  - botocore
  - fsspec >=0.8.0
  - python >=3.6
```

And in turn, aiobotocore has a very restrictive `botocore` dependency:
```
aiobotocore 1.1.2 py_0
----------------------
file name   : aiobotocore-1.1.2-py_0.tar.bz2
name        : aiobotocore
version     : 1.1.2
build       : py_0
build number: 0
size        : 39 KB
license     : Apache-2.0
subdir      : noarch
url         : https://conda.anaconda.org/conda-forge/noarch/aiobotocore-1.1.2-py_0.tar.bz2
md5         : 3954af1cdc59a342befe2069f3df51ac
timestamp   : 2020-10-07 10:41:30 UTC
dependencies: 
  - aiohttp >=3.3.1
  - aioitertools >=0.5.1
  - botocore >=1.17.44,<1.17.45
  - python >=3.6
  - wrapt >=1.10.10
```

**Comment by ocefpaf** [comment]
@scottyhq sorry for getting here too late but all these boto-stack packages are tightly coupled and with restrictive pins. We tried to loosen them a bit in the past but upstream devs were not happy. Now we specify exactly what is in the package metadata.

TL;DR what you did is the only way to ensure the packages you want are in the solve, or force the solve to error otherwise.

**Comment by scottyhq** [comment]
While we're now installing s3fs>0.5, I wouldn't say this issue is fully resolved since we'd also like newer versions of botocore&boto3 in the environment. Here is related discussion on s3fs https://github.com/dask/s3fs/issues/357
